var/spack/repos/builtin/packages/intel-oneapi-mpi/package.py

```diff
@@ -50,3 +54,43 @@ def libs(self):
             ldir = find_libraries('*', root=lib_path, shared=True, recursive=False)
             libs += ldir
         return libs
+
+    def _join_prefix(self, path):
+        return join_path(self.prefix, 'mpi', 'latest', path)
+
+    def _ld_library_path(self):
+        dirs = ['lib',
+                'lib/release',
+                'libfabric/lib']
+        for dir in dirs:
+            yield self._join_prefix(dir)
+
+    def _library_path(self):
+        dirs = ['lib',
+                'lib/release',
+                'libfabric/lib']
+        for dir in dirs:
+            yield self._join_prefix(dir)
+
+    def install(self, spec, prefix):
+        super(IntelOneapiMpi, self).install(spec, prefix)
+
+        # need to patch libmpi.so so it can always find libfabric
+        libfabric_rpath = self._join_prefix('libfabric/lib')
+        for lib_version in ['debug', 'release', 'release_mt', 'debug_mt']:
+            file = self._join_prefix('lib/' + lib_version + '/libmpi.so')
+            subprocess.call(['patchelf', '--set-rpath', libfabric_rpath, file])
+
+    def setup_run_environment(self, env):
+        env.prepend_path('PATH', self._join_prefix('bin'))
+        env.prepend_path('CPATH', self._join_prefix('include'))
+        for dir in self._library_path():
+            env.prepend_path('LIBRARY_PATH', dir)
+        for dir in self._ld_library_path():
+            env.prepend_path('LD_LIBRARY_PATH', dir)
+        # so wrappers know where MPI lives
+        mpi_root = join_path(prefix, 'mpi', 'latest')
+        env.set('I_MPI_ROOT', mpi_root)
+        # set these so that wrappers can find libfabric providers
+        env.set('FI_PROVIDER_PATH', '/usr/lib64/libfabric')
```

Comment:
This appears to be hardcoded (users could place these files anywhere). It looks like this might be a dependency of `intel-oneapi-mpi` that we aren't currently accounting for. Or is this intended to help with externally-installed `intel-oneapi-mpi`?

Reply:
This looks to have been a temporary fix before I got the rpath set. An additional value for the path is set to ".../mpi/latest/libfabric/lib/prov" but I need to dig a bit and see where. The hardcoded path should go away.

Reply:
> This looks to have been a temporary fix before I got the rpath set. An additional value for the path is set to ".../mpi/latest/libfabric/lib/prov" but I need to dig a bit and see where. The hardcoded path should go away.

Do you plan to address this in this PR?

Reply:
Yes. It is innocuous as-is, but there shouldn't be a hard coded path and I will look through to remember why I did this, and verify that it's no longer necessary. If you need to merge for a release, it shouldn't hurt anything, otherwise I'd need a day or two to sort it out.